components/ws-manager/pkg/manager/ws_daemon_controller.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package manager
+
+import (
+	"context"
+
+	"github.com/gitpod-io/gitpod/ws-manager/pkg/manager/internal/grpcpool"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// WSDamonReconciler reconciles ws-daemon pods
+type WSDamonReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+
+	WSDaemonPool *grpcpool.Pool
+}
+
+func (r *WSDamonReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	var pod corev1.Pod
+	err := r.Client.Get(ctx, req.NamespacedName, &pod)
+	if err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if !pod.DeletionTimestamp.IsZero() {
+		_ = r.WSDaemonPool.Remove(pod.Spec.NodeName)
```

Comment:
Should we check the error here? If error happens, keep reconcile it.

Reply:
done